independent-projects/arc/tests/src/test/java/io/quarkus/arc/test/event/qualifier/EventDefaultQualifierTest.java

```diff
@@ -0,0 +1,159 @@
+import io.quarkus.arc.test.ArcTestContainer;
+
+public class EventDefaultQualifierTest {
+
+    @RegisterExtension
+    public ArcTestContainer container = ArcTestContainer.builder().beanClasses(ObservingBean.class).build();
+
+    @Test
+    public void testDefaultQualifierPresent()
+            throws ExecutionException, InterruptedException, TimeoutException {
+        BeanManager bm = Arc.container().beanManager();
+        ObservingBean bean = Arc.container().select(ObservingBean.class).get();
+        bean.reset();
+
+        Set<Annotation> expectedQualifiers = Set.of(Any.Literal.INSTANCE, Default.Literal.INSTANCE);
+
+        // just get event fire right away - @Default should be included
+        bm.getEvent().fire(new Payload());
+        Assertions.assertEquals(1, bean.getDefaultObjectNotified());
+        Assertions.assertEquals(1, bean.getDefaultPayloadNotified());
+        Assertions.assertEquals(expectedQualifiers, bean.getDefaultObjectQualifiers());
+        Assertions.assertEquals(expectedQualifiers, bean.getDefaultPayloadQualifiers());
+
+        // select Payload and fire - @Default should be included
+        bm.getEvent().select(Payload.class).fire(new Payload());
+        Assertions.assertEquals(2, bean.getDefaultObjectNotified());
+        Assertions.assertEquals(2, bean.getDefaultPayloadNotified());
+        Assertions.assertEquals(expectedQualifiers, bean.getDefaultObjectQualifiers());
+        Assertions.assertEquals(expectedQualifiers, bean.getDefaultPayloadQualifiers());
+
+        // select Payload and explicitly add @Any qualifier, then fire - @Default should *not* be included
+        // therefore no notifications should occur
+        bm.getEvent().select(Payload.class, Any.Literal.INSTANCE).fire(new Payload());
+        Assertions.assertEquals(2, bean.getDefaultObjectNotified());
+        Assertions.assertEquals(2, bean.getDefaultPayloadNotified());
+
+        // same in async variant
+        // just get event fire right away - @Default should be included
+        bm.getEvent().fireAsync(new Payload()).toCompletableFuture().get(2, TimeUnit.SECONDS);
+        Assertions.assertEquals(1, bean.getDefaultObjectAsyncNotified());
+        Assertions.assertEquals(1, bean.getDefaultPayloadAsyncNotified());
+        Assertions.assertEquals(expectedQualifiers, bean.getDefaultObjectAsyncQualifiers());
+        Assertions.assertEquals(expectedQualifiers, bean.getDefaultPayloadAsyncQualifiers());
+
+        // select Payload and fire - @Default should be included
+        bm.getEvent().select(Payload.class).fireAsync(new Payload()).toCompletableFuture().get(2, TimeUnit.SECONDS);
+        Assertions.assertEquals(2, bean.getDefaultObjectAsyncNotified());
+        Assertions.assertEquals(2, bean.getDefaultPayloadAsyncNotified());
+        Assertions.assertEquals(expectedQualifiers, bean.getDefaultObjectAsyncQualifiers());
+        Assertions.assertEquals(expectedQualifiers, bean.getDefaultPayloadAsyncQualifiers());
+
```

Comment:
I know I'm being picky, but there are 3 tests for sync events and only 2 for async events. The test for async event with explicitly provided `@Any` qualifier seems missing.

Reply:
We use the same logic for both, sync and async events but I get your point,
I've added the respective async variant.